src/main/java/com/faforever/client/chat/AbstractChatTabController.java

```diff
@@ -431,6 +466,14 @@ private void sendAction(final TextInputControl messageTextField, final String te
   }
 
   protected void onChatMessage(ChatMessage chatMessage) {
+    if(chatMessage.username().equals(playerService.getCurrentPlayer().getUsername())) {
+      if(userMessageHistory.size() >= 50){
+        userMessageHistory.remove(0);
+        userMessageHistory.add(chatMessage);
+      } else {
+        userMessageHistory.add(chatMessage);
+      }
+    }
```

Comment:
One thing I realized is that this is probably better off put in the send message method. Then you don't have to check the username